pages/create-issue.tsx

```diff
@@ -34,25 +34,22 @@ export default function PageCreateIssue() {
   }, [])
 
   const allow = async (evt) => {
-    evt.preventDefault();        
+    evt.preventDefault();
+    let beproAddress       
     await BeproService.login()
       .then(() => setLoadingAttributes(true))
-      .catch((error) => handleErrorAndLoading(error));
-    const beproAddress = await BeproService.getAddress();
-    const payload = {
-      amount: issueAmount.floatValue,
-      address: beproAddress,
-    };
-    await BeproService.network
-      .approveTransactionalERC20Token()
-      .then(async () => {
-        await BeproService.network
-        .isApprovedTransactionalToken(payload)
-        .then(() => {
-          setAllowedTransaction(true);
-          setLoadingAttributes(false);
-        })
-        .catch((error) => handleErrorAndLoading(error))
+      .then(() => { beproAddress = BeproService.getAddress() })
```

Comment:
turn this around, do approveTransactionaERC20Token() before .getAddress():

```ts
setLoadingAttributes(true) // BEFORE ANYTHING SET LOADER TO TRUE
await BepropService.login()
  .then(BeproService.network.approveTransactionalERC20Token) // approve transaction
  .then(BeproService.getAddress) // get and return the address 
  .then(address => BeproService.network.isApprovedTransactionalToken({address, amount: issueAmount.floatValue})) // you don't need a const for a value that you'll use right after, just provide the payload as an argument instead of as a const
  .then(transaction => {
    setAllowedTransaction(transaction)
    setLoadingAttributes(false)
  })
  .catch((error) => handleErrorAndLoading(error));